Open Judge System/Web/OJS.Web/Views/Shared/_AdvancedSubmissionsGridPartial.cshtml

```diff
@@ -93,4 +93,24 @@
             counter++;
         });
     }
+</script>
+<script>
+    function retest(id) {
+        var __RequestVerificationToken = $("[name='__RequestVerificationToken']").val();
+        $.ajax({
+            type: "post",
+            dataType: "html",
+            url: '/Administration/Submissions/Retest/' + id,
+            data: {
+                __RequestVerificationToken
+            },
```

Comment:
Extract common function that does the post if there is none.

Reply:
The ajax post takes in too many variables, which could be different. It can have different data types, a different url, there might be other data that you want to pass, you have the success and error behaviors. Imo there is no point in extracting it in a different file.